File: esap-core/src/main/java/ru/javavlsu/kb/esap/dto/notifications/PatientCreatedEvent.java
package ru.javavlsu.kb.esap.dto.notifications;

import ru.javavlsu.kb.esap.model.Patient;

/**
 * PatientCreatedEvent 24.07.2026 thewyolar
 * Copyright (c) 2026.
 */
public record PatientCreatedEvent(
        String email,
        String firstName,
        String login,
        String password,
        String clinicName
) {
    public static PatientCreatedEvent from(Patient patient) {
        return new PatientCreatedEvent(
                patient.getEmail(),
                patient.getFirstName(),
                patient.getLogin(),
                patient.getPassword(),
                patient.getClinic().getName()
        );
    }
}

File: notification-service/src/main/kotlin/ru/javavlsu/kb/notificationservice/dto/PatientCreatedEvent.kt
package ru.javavlsu.kb.notificationservice.dto

/**
 * UserCreatedEvent 24.07.2026 thewyolar
 * Copyright (c) 2026.
 */
data class PatientCreatedEvent(
    val email: String,
    val firstName: String,
    val login: String,
    val password: String,
    val clinicName: String
)
